sky/backends/cloud_vm_ray_backend.py

```diff
@@ -4713,6 +4717,18 @@ def _check_existing_cluster(
                             'a new cluster with the desired ports open.')
             if all_ports:
                 to_provision = to_provision.copy(ports=all_ports)
+            # Docker login should always be the same for all resources, since
+            # it's set from envs.
+            for resource in task.resources:
+                assert (resource.docker_login_config ==
+                        one_task_resource.docker_login_config), (
+                            resource.docker_login_config,
+                            one_task_resource.docker_login_config)
+            # If we have docker login config in the new task, override the
+            # existing resources to pick up new credentials.
```

Comment:
Can we elaborate a bit more for why we need the override, e.g. user updates the docker login credential for previously wrongly configured the credential?